src/Castle.Core.Tests/Castle.Core.Tests.csproj

```diff
@@ -38,8 +39,10 @@
 	</ItemGroup>
 
 	<ItemGroup>
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
```

Comment:
Interesting to see `Microsoft.NET.Test.Sdk` is versioned 16.2.0. Is this a package just for Visual Studio with a name making it sound like it is for all .NET CLI unit testing.

Reply:
AFAIK, the package version isn't directly tied to a particular version of VS. But I have never been too clear what exactly requires this package... maybe VS, maybe VSTest. I might run some experiments to find out. Wil report back my findings if & when I do so.

Reply:
@jonorossi, while I don't understand the whole theory behind that package, it indeed enables `dotnet test`! In fact, we can use all of `dotnet restore|build|test` now, if we make one further addition in `common.props` that looks roughly like the following (as per https://github.com/Microsoft/msbuild/issues/1333#issuecomment-296346352):

```xml
<PropertyGroup>
  <FrameworkPathOverride Condition="'$(TargetFramework)|$(OS)' == 'net35|Windows'">$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\Framework\.NETFramework\v3.5\Profile\Client</FrameworkPathOverride>
  <FrameworkPathOverride Condition="'$(TargetFramework)|$(OS)' == 'net35|Unix'">$(MONO_DIR)/../lib/mono/2.0-api</FrameworkPathOverride>
</PropertyGroup>
```

(We already do something similar for .NET 3.5 compilation using Mono. I've only verified `dotnet` tooling for Windows, the Mono/Linux override will need some more work. But essentially, the `MONO_DIR` variable can be set on the shell using `export MONO_DIR=$(dirname $(which mono))`, then passed to the build.)